misc/python/materialize/checks/mzcompose_actions.py

```diff
@@ -65,19 +65,6 @@ def execute(self, e: Executor) -> None:
         image = f"materialize/materialized:{self.tag}" if self.tag is not None else None
         print(f"Starting Mz using image {image}, mz_service {self.mz_service}")
 
-        # Before #22790, this parameter needs to be set to False to avoid a panic
-        # Importantly, the environmentd must boot with the flag set to False, so we have to set it here
-        # By the time ConfigureMz() arrives to run ALTER SYSTEM SET, it is already too late
-        # to disable it, as problematic introspection dataflows have been created already
-        if (
-            self.tag
-            and "enable_specialized_arrangements"
-            not in self.additional_system_parameter_defaults
-        ):
-            self.additional_system_parameter_defaults[
-                "enable_specialized_arrangements"
-            ] = "false"
-
```

Comment:
Removing all the CI special casing works because this feature flag was always disabled by default on main.